src/test/java/hudson/plugin/versioncolumn/VersionMonitorTest.java

```diff
@@ -0,0 +1,202 @@
+        Computer computer = mock(Computer.class);
+
+        when(computer.getChannel()).thenReturn(null);
+
+        String result = descriptor.monitor(computer);
+
+        assertEquals("unknown-version", result);
+        verify(computer, never()).setTemporarilyOffline(anyBoolean(), any());
+    }
+
+    @Test
+    public void testMonitor_SameVersion() throws IOException, InterruptedException {
+        VersionMonitor.DescriptorImpl descriptor = new VersionMonitor.DescriptorImpl();
+        Computer computer = mock(Computer.class);
+        VirtualChannel channel = mock(VirtualChannel.class);
+
+        when(computer.getChannel()).thenReturn(channel);
+        when(channel.call(ArgumentMatchers.<MasterToSlaveCallable<String, IOException>>any()))
+                .thenReturn(Launcher.VERSION);
+        when(computer.isOffline()).thenReturn(false);
+
+        String result = descriptor.monitor(computer);
+
+        assertEquals(Launcher.VERSION, result);
+        verify(computer, never()).setTemporarilyOffline(anyBoolean(), any());
+    }
+
+    @Test
+    public void testMonitor_DifferentVersion_NotIgnored() throws IOException, InterruptedException {
+        VersionMonitor.DescriptorImpl descriptor = spy(new VersionMonitor.DescriptorImpl());
+        doReturn(false).when(descriptor).isIgnored(); // Ensure isIgnored returns false
+
+        Computer computer = mock(Computer.class);
+        VirtualChannel channel = mock(VirtualChannel.class);
+        String differentVersion = "different-version";
+
+        when(computer.getChannel()).thenReturn(channel);
+        when(channel.call(ArgumentMatchers.<MasterToSlaveCallable<String, IOException>>any()))
+                .thenReturn(differentVersion);
+        when(computer.isOffline()).thenReturn(false);
+        when(computer.getName()).thenReturn("TestComputer");
+
+        String result = descriptor.monitor(computer);
+
+        assertEquals(differentVersion, result);
+        verify(computer).setTemporarilyOffline(eq(true), any(VersionMonitor.RemotingVersionMismatchCause.class));
+    }
+
+    @Test
+    public void testMonitor_DifferentVersion_Ignored() throws IOException, InterruptedException {
+        VersionMonitor.DescriptorImpl descriptor = spy(new VersionMonitor.DescriptorImpl());
+        doReturn(true).when(descriptor).isIgnored(); // Ensure isIgnored returns true
+
+        Computer computer = mock(Computer.class);
+        VirtualChannel channel = mock(VirtualChannel.class);
+        String differentVersion = "different-version";
+
+        when(computer.getChannel()).thenReturn(channel);
+        when(channel.call(ArgumentMatchers.<MasterToSlaveCallable<String, IOException>>any()))
+                .thenReturn(differentVersion);
+        when(computer.isOffline()).thenReturn(false);
+
+        String result = descriptor.monitor(computer);
+
+        assertEquals(differentVersion, result);
+        verify(computer, never()).setTemporarilyOffline(anyBoolean(), any());
+    }
```

Comment:
These newly-added tests are failing on recent cores as of jenkinsci/jenkins#9855.